src/detect-ssh-proto.c

```diff
@@ -47,159 +47,85 @@
 #include "app-layer-ssh.h"
 #include "detect-ssh-proto.h"
 
-#define KEYWORD_NAME "ssh_proto"
+#define KEYWORD_NAME "ssh.proto"
+#define KEYWORD_NAME_LEGACY "ssh_proto"
 #define KEYWORD_DOC "ssh-keywords.html#ssh-proto"
 #define BUFFER_NAME "ssh_protocol"
 #define BUFFER_DESC "ssh protocol field"
 static int g_buffer_id = 0;
 
-/** \brief SSH Protocol Mpm prefilter callback
- *
- *  \param det_ctx detection engine thread ctx
- *  \param p packet to inspect
- *  \param f flow to inspect
- *  \param txv tx to inspect
- *  \param pectx inspection context
- */
-static void PrefilterTxSshRequestProtocol(DetectEngineThreadCtx *det_ctx,
-        const void *pectx,
-        Packet *p, Flow *f, void *txv,
-        const uint64_t idx, const uint8_t flags)
+static InspectionBuffer *GetSshData(DetectEngineThreadCtx *det_ctx,
+        const DetectEngineTransforms *transforms, Flow *_f,
+        const uint8_t _flow_flags, void *txv, const int list_id)
 {
     SCEnter();
 
-    const MpmCtx *mpm_ctx = (MpmCtx *)pectx;
-    const SshState *ssh_state = txv;
-
-    if (ssh_state->cli_hdr.proto_version == NULL)
-        return;
-
-    uint32_t buffer_len = strlen((char *)ssh_state->cli_hdr.proto_version);
-    const uint8_t *buffer = ssh_state->cli_hdr.proto_version;
-
-    if (buffer_len >= mpm_ctx->minlen) {
-        (void)mpm_table[mpm_ctx->mpm_type].Search(mpm_ctx,
-                &det_ctx->mtcu, &det_ctx->pmq, buffer, buffer_len);
-    }
-}
+    InspectionBuffer *buffer = InspectionBufferGet(det_ctx, list_id);
 
-static int PrefilterTxSshRequestProtocolRegister(DetectEngineCtx *de_ctx,
-        SigGroupHead *sgh, MpmCtx *mpm_ctx)
-{
-    SCEnter();
-
-    int r = PrefilterAppendTxEngine(de_ctx, sgh, PrefilterTxSshRequestProtocol,
-        ALPROTO_SSH, SSH_STATE_BANNER_DONE,
-        mpm_ctx, NULL, KEYWORD_NAME " (request)");
-    return r;
-}
-
-/** \brief SSH Protocol Mpm prefilter callback
- *
- *  \param det_ctx detection engine thread ctx
- *  \param p packet to inspect
- *  \param f flow to inspect
- *  \param txv tx to inspect
- *  \param pectx inspection context
- */
-static void PrefilterTxSshResponseProtocol(DetectEngineThreadCtx *det_ctx,
-        const void *pectx,
-        Packet *p, Flow *f, void *txv,
-        const uint64_t idx, const uint8_t flags)
-{
-    SCEnter();
+    if (buffer->inspect == NULL) {
+        uint8_t *protocol = NULL;
+        SshState *ssh_state = (SshState *) txv;
 
-    const MpmCtx *mpm_ctx = (MpmCtx *)pectx;
-    const SshState *ssh_state = txv;
+        if (_flow_flags & STREAM_TOSERVER)
```

Comment:
the leading underscore is something adopted from Rust, where it's a way to tell the compiler the function argument is unused. Since here it is being used, please remove the leading underscore.

Reply:
I fixed this up myself.